api/src/main/java/jakarta/data/Restriction.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+import java.util.List;
+
+public interface Restriction<T> {
+    boolean isNegated();
+
+    interface Basic<T> extends Restriction<T> {
+        Operator comparison();
+
+        String field();
+
+        Object value();
+    }
+
+    interface Composite<T> extends Restriction<T> {
+        List<Restriction<T>> restrictions();
+
+        Type type();
+
+        enum Type {
+            ALL,
+            ANY
+        }
+    }
+
+    interface Text<T> extends Basic<T> {
+        Restriction<T> ignoreCase();
+
+        boolean isAnyCase();
```

Comment:
```suggestion
        boolean isCaseSensitive();
```

Reply:
Lacking a better name, we can use Gavin's suggestion of isCaseSensitive.  This method is here for the Jakarta Data provider rather than the user, and we can easily add a disclaimer in the JavaDoc that if a database is incapable of case sensitive comparison, it will ignore a value of true.